src/parser/symbol_table.cpp

```diff
@@ -621,6 +637,12 @@ bool SymbolTable::Implementation::bindWithOverloading(const string& name,
     // the symbol manager.
     if (prev_bound_obj != obj)
     {
+      // If the previous overloaded symbol was a sort, we fail unconditionally
+      // in this case.
```

Comment:
It's not very clear what is happening here, because it is not very clear what d_dummySortTerms is for.